### PR TITLE
Move API conditions and phases into common api file

### PR DESCRIPTION
**Pull request by mpryc**
Change which moves Conditions and Phases into separate nonadmin_types.go that will be used across all other controllers within NonAdmin project. This ensures we are making them part of the API to match them closer to the design.

## Why the changes were made

These changes were introduced to improve the consistency of the NonAdmin project by centralizing the Conditions and Phases. Moving them to a dedicated nonadmin_types.go file provides a single source of truth within API, ensuring that any controller or component working with NonAdminBackup and NonAdminRestore objects can reference these conditions and phases in a consistent way and aligned with design documents and diagrams, ensuring that they are part of the standard data structure shared across the project.

## How to test the changes made

Moving some constants and extracting common strings. Followed by running tests and linters:

```console
$ make simulation-test
$ make lint
```

### CODE REVIEW

**Comment by mpryc** [comment]
I'd like to keep this a separate PR and pull out from other PRs such as #115 or #128.

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by kaovilai — APPROVED** [review]
/lgtm

**Comment by mpryc** [comment]
Depends on https://github.com/openshift/oadp-operator/pull/1605

File: api/v1alpha1/nonadmin_types.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// array of the conditions through which the NonAdminController has or has not passed
+const (
+	NonAdminConditionAccepted NonAdminCondition = "Accepted"
+	NonAdminConditionQueued   NonAdminCondition = "Queued"
+	NonAdminConditionDeleting NonAdminCondition = "Deleting"
+)
+
+// QueueInfo holds the queue position for a specific operation.
+type QueueInfo struct {
+	// estimatedQueuePosition is the number of backups ahead in the queue (0 if not queued)
```
Comment (mateusoliveira43) [inline api/v1alpha1/nonadmin_types.go:50]: this is going to be also used by restore? need to change comment

Comment (mpryc) [reply api/v1alpha1/nonadmin_types.go]: @mateusoliveira43 you are right, updated to be "operations" as you suggested in other PR, I've moved the description of the actual backup to the backup part of it, similarly restore will have it:

https://github.com/migtools/oadp-non-admin/pull/130/files#diff-1fc992498fb761a2b0d175e637f5f945c2eb932ffacb0713d8724f4c14054c31R86-R89

Comment (mpryc) [reply api/v1alpha1/nonadmin_types.go]: Ah forgot to add the change to commit. thx.

**Review by mateusoliveira43 — APPROVED** [review]
/lgtm

**Pull request merged**